example/tests/test_openapi.py

```diff
@@ -0,0 +1,176 @@
+
+from example import models, serializers, views
+
+
+def create_request(path):
+    factory = RequestFactory()
+    request = Request(factory.get(path))
+    return request
+
+
+def create_view(view_cls, method, request):
+    generator = SchemaGenerator()
+    view = generator.create_view(view_cls.as_view(), method, request)
+    return view
+
+
+def create_view_with_kw(view_cls, method, request, initkwargs):
+    generator = SchemaGenerator()
+    view = generator.create_view(view_cls.as_view(initkwargs), method, request)
+    return view
+
+
+def test_path_without_parameters(snapshot):
+    path = '/authors/'
+    method = 'GET'
+
+    view = create_view_with_kw(
+        views.AuthorViewSet,
+        method,
+        create_request(path),
+        {'get': 'list'}
+    )
+    inspector = AutoSchema()
+    inspector.view = view
+
+    operation = inspector.get_operation(path, method)
+    snapshot.assert_match(json.dumps(operation, indent=2, sort_keys=True))
+
+
+def test_path_with_id_parameter(snapshot):
+    path = '/authors/{id}/'
+    method = 'GET'
+
+    view = create_view_with_kw(
+        views.AuthorViewSet,
+        method,
+        create_request(path),
+        {'get': 'retrieve'}
+    )
+    inspector = AutoSchema()
+    inspector.view = view
+
+    operation = inspector.get_operation(path, method)
+    snapshot.assert_match(json.dumps(operation, indent=2, sort_keys=True))
+
+
+def test_post_request(snapshot):
+    method = 'POST'
+    path = '/authors/'
+
+    view = create_view_with_kw(
+        views.AuthorViewSet,
+        method,
+        create_request(path),
+        {'post': 'create'}
+    )
+    inspector = AutoSchema()
+    inspector.view = view
+
+    operation = inspector.get_operation(path, method)
+    snapshot.assert_match(json.dumps(operation, indent=2, sort_keys=True))
+
+
+def test_patch_request(snapshot):
+    method = 'PATCH'
+    path = '/authors/{id}'
+
+    view = create_view_with_kw(
+        views.AuthorViewSet,
+        method,
+        create_request(path),
+        {'patch': 'update'}
+    )
+    inspector = AutoSchema()
+    inspector.view = view
+
+    operation = inspector.get_operation(path, method)
+    snapshot.assert_match(json.dumps(operation, indent=2, sort_keys=True))
+
+
+def test_delete_request(snapshot):
+    method = 'DELETE'
+    path = '/authors/{id}'
+
+    view = create_view_with_kw(
+        views.AuthorViewSet,
+        method,
+        create_request(path),
+        {'delete': 'delete'}
+    )
+    inspector = AutoSchema()
+    inspector.view = view
+
+    operation = inspector.get_operation(path, method)
+    snapshot.assert_match(json.dumps(operation, indent=2, sort_keys=True))
+
+
+@override_settings(REST_FRAMEWORK={
+    'DEFAULT_SCHEMA_CLASS': 'rest_framework_json_api.schemas.openapi.AutoSchema'})
+def test_schema_construction():
+    """Construction of the top level dictionary."""
+    patterns = [
+        url(r'^authors/?$', views.AuthorViewSet.as_view({'get': 'list'})),
```

Comment:
why not using path() or re_path()?

Reply:
I believe these were only [added to django in 2.0](https://docs.djangoproject.com/en/2.2/releases/2.0/#simplified-url-routing-syntax). We still support django 1.11 LTS.